src/boot/OfflineNoticeProvider.js

```diff
@@ -0,0 +1,454 @@
+        LayoutAnimation.configureNext({
+          ...LayoutAnimation.Presets.easeInEaseOut,
+
+          // Enter slowly to give bad, possibly unexpected news. Leave quickly
+          // to give good, hoped-for news.
+          duration: newValue ? 1000 : 300,
+        });
+      }
+      return newValue;
+    });
+  }, [isOnline, shouldShowUncertaintyNotice, _]);
+
+  const prevIsOnline = usePrevious(isOnline);
+  useConditionalEffect(
+    useCallback(() => {
+      // TODO(rn-68): Use announceForAccessibilityWithOptions to queue the
+      //   "connection restored" announcement behind in-progress announcements
+      AccessibilityInfo.announceForAccessibility(
+        /* prettier-ignore */
+        isOnline === false
+          // Mention Zulip so this doesn't sound like an announcement from
+          // the OS. We don't speak for the OS, and the OS might disagree
+          // about the connectivity state, e.g., because we're wrong about
+          // Zulip's connectivity or because a connection problem somehow
+          // affects Zulip but not other apps.
+          //
+          // Our UI element for the notice shouldn't have to mention Zulip
+          // because it's already clear that it's from Zulip: it's part of
+          // the UI we draw, as traversed visually or by a screen reader.
+          // The OS should make it clear when you're traversing a given
+          // app's UI, e.g., so an app can't trick you into giving it
+          // sensitive data that you meant for the OS or another app.
+          ? _('Zulip offline.')
+          : _('Zulip back online.'),
+      );
+    }, [isOnline, _]),
+    // Ignore shouldShowUncertaintyNotice; it basically exists to give us
+    // debugging clues when NetInfo misbehaves, and isn't very informative
+    // or actionable to the user.
+    (prevIsOnline === true && isOnline === false) || (prevIsOnline === false && isOnline === true),
+  );
+
+  const styles = useMemo(
+    () =>
+      createStyleSheet({
+        flex1: { flex: 1 },
+        noticeSurface: {
+          position: 'absolute',
+
+          // Whether the notice is visible or tucked away above the window.
+          //
+          // (Just as we discovered in 3fa7a7f10 with the lightbox, it seems
+          // the Animated API wouldn't let us do a translate-transform
+          // animation with a percentage, like '100%'; that's issue
+          //   https://github.com/facebook/react-native/issues/13107 .
+          // So we use LayoutAnimation, which is probably better anyway
+          // because it lets us animate layout changes at the native layer,
+          // and so won't drop frames when the JavaScript thread is busy.)
+          ...(isNoticeVisible ? { top: 0 } : { bottom: windowHeight }),
+
+          zIndex: 1,
+          width: windowWidth,
+
+          // If changing, also change the status bar color in
+          // OfflineNoticePlaceholder.
+          backgroundColor: HALF_COLOR,
```

Comment:
> [in the status quo] this banner isn't overlain on anything so the HALF_COLOR gray only mixes with the app's overall background color.

Yeah, that makes sense. That suggests that two colors are used in the status quo, because there are two overall background colors: a light one for light mode, a dark one for dark mode. I can make an opaque background that matches our current two colors exactly, or we could ask Vlad if he'd like to specify the background color and perhaps the text color too.